src/safeposix/syscalls/fs_calls.rs

```diff
@@ -180,27 +261,35 @@ impl Cage {
         fd //open returns the opened file descriptor
     }
 
-    //------------------MKDIR SYSCALL------------------
-    // Description
-    // The mkdir_syscall() creates a new directory named by the path name pointed to by a path as the input parameter in the function.
-    // The mode of the new directory is initialized from the "mode" provided as the input parameter in the function.
-    // The newly created directory is empty with size 0 and is associated with a new inode of type "DIR".
-    // On successful completion, the timestamps for both the newly formed directory and its parent are updated along with their linkcounts.
-    
-    // Function Arguments
-    // The mkdir_syscall() receives two arguments:
-    // 1. Path - This represents the path at which the new directory will be created.
-    //           For example: "/parentdir/dir" represents the new directory name as "dir", which will be created at this path (/parentdir/dir).
-    // 2. Mode - This represents the permission of the newly created directory. 
-    //           The general mode used is "S_IRWXA": which represents the read, write, and search permissions on the new directory. 
+    /// ## ------------------MKDIR SYSCALL------------------
+    /// 
+    /// ### Description
+    /// 
+    /// The mkdir_syscall() creates a new directory named by the path name pointed to by a path as the input parameter in the function.
```

Comment:
Text with _ or other special characters is often treated as markdown by rustdoc.   I think using backquotes around function / variable names fixes this...